resources/eni.rb

```diff
@@ -0,0 +1,121 @@
+require 'aws-sdk'
+require 'net/http'
+
+default_action :create_and_attach
+
+property :description,          String
+property :subnet_id,            String, required: true
+property :private_ip_address,   String
+property :security_groups,      Array
+property :network_interface_id, String
+
+action :create_and_attach do
+  nic_id_to_attach = network_interface_id || create
+
+  if attached_nic_ids.include? nic_id_to_attach
+    Chef::Log.debug "NIC with ID #{nic_id_to_attach} is already attached to #{instance_id}"
+  else
+    converge_by "Attach NIC #{nic_id_to_attach} to #{instance_id} at index #{next_device_index}" do
+      ec2.attach_network_interface(
+        network_interface_id: nic_id_to_attach,
+        instance_id: instance_id,
+        device_index: next_device_index
+      )
+    end
+  end
+end
+
+action :create do
+  create
+end
+
+action :delete do
+  if existing_nic
+    converge_by "Delete the NIC with ID #{existing_nic.network_interface_id}" do
+      ec2.delete_network_interface network_interface_id: existing_nic.network_interface_id
+    end
+  elsif network_interface_id
+    Chef::Log.debug "NIC with ID #{network_interface_id} does not exist"
+  else
+    Chef::Log.debug "NIC with description \"#{description}\" does not exist"
+  end
+end
+
+private
```

Comment:
All these can be added to the `action_class` block rather than adding them to the 'outer' resource class.

Reply:
https://docs.chef.io/custom_resources/#action_class

Reply:
Thanks @bmhughes! I did a big refactor in #44, so once that merges, and I rebase #43 & get it merged, I’ll come back to this one.